### ISSUE TITLE
Deep linking through web url doesn't open my expo app in android

### ISSUE THREAD

**Issue by DSp4wN** [expo/expo#29210]
### Minimal reproducible example

Any latest template with following app json config

### What platform(s) does this occur on?

Android

### Did you reproduce this issue in a development build?

Yes

### Summary

So I followed https://docs.expo.dev/guides/deep-linking/ to implement deep linking my expo react native webview application, now I have properly configured the digital asset link json given in the google play console in my web server and it also shows that the deep linking is completely functional.

![image](https://github.com/expo/expo/assets/48817540/c3aefaea-c5d9-4471-a8ee-44f246382c5f)

### Environment

expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 14.5
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 20.12.2 - /usr/local/bin/node
      Yarn: 1.22.19 - /usr/local/bin/yarn
      npm: 10.5.0 - /usr/local/bin/npm
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.5, iOS 17.5, macOS 14.5, tvOS 17.5, visionOS 1.2, watchOS 10.5
    IDEs:
      Android Studio: 2022.2 AI-222.4459.24.2221.9971841
      Xcode: 15.4/15F31d - /usr/bin/xcodebuild
    npmPackages:
      expo: 51.0.8 => 51.0.8 
      react: 18.2.0 => 18.2.0 
      react-native: 0.74.1 => 0.74.1 
    npmGlobalPackages:
      eas-cli: 9.0.3
    Expo Workflow: bare

### Expo Doctor Diagnostics

✔ Check Expo config for common issues
✔ Check package.json for common issues
✔ Check dependencies for packages that should not be installed directly
✔ Check for common project setup issues
✖ Check native tooling versions
✔ Check npm/ yarn versions
✔ Check for issues with metro config
✔ Check for legacy global CLI installed locally
✔ Check that native modules do not use incompatible support packages
✔ Check Expo config (app.json/ app.config.js) schema
✔ Check that native modules use compatible support package versions for installed Expo SDK
✖ Check that packages match versions required by installed Expo SDK

Detailed check results:

Cocoapods version check failed. Cocoapods may not be installed or there may be an issue with your Cocoapods installation. Installing version 1.15.2 or higher is recommended.

The following packages should be updated for best compatibility with the installed expo version:
  expo@51.0.8 - expected version: ~51.0.9
  expo-camera@15.0.9 - expected version: ~15.0.10
  expo-constants@16.0.1 - expected version: ~16.0.2
  expo-notifications@0.28.3 - expected version: ~0.28.4

**Comment by brentvatne** [comment]
you need to use a development build for these features :)

**Comment by DSp4wN** [comment]
@brentvatne development build? we cannot use it in playstore build?

**Comment by brentvatne** [comment]
@DSp4wN - that works fine too. it just doesn't work in expo go.

**Comment by brentvatne** [comment]
if you are encountering this in a production build you can check out resources like https://doordash.engineering/2022/01/25/your-deep-links-might-be-broken-web-intents-and-android-12/ to learn how to debug the issue

**Comment by brentvatne** [comment]
@DSp4wN - you can run `npx expo prebuild -p android` to inspect the `android` directory and verify that the config is applied correctly

**Comment by DSp4wN** [comment]
Yes I did prebuild after every step and this is how my intent filters look in android manifest @brentvatne 

<activity android:name=".MainActivity" android:configChanges="keyboard|keyboardHidden|orientation|screenSize|screenLayout|uiMode" android:launchMode="singleTask" android:windowSoftInputMode="adjustResize" android:theme="@style/Theme.App.SplashScreen" android:exported="true" android:screenOrientation="portrait">
      <intent-filter>
        <action android:name="android.intent.action.MAIN"/>
        <category android:name="android.intent.category.LAUNCHER"/>
      </intent-filter>
      <intent-filter android:autoVerify="true" data-generated="true">
        <action android:name="android.intent.action.VIEW"/>
        <data android:scheme="https"/>
        <data android:scheme="exp+myappnetworks"/>
        <data android:scheme="com.myappmobileapp"/>
        <data android:scheme="https" android:host="www.myapp.in" android:pathPrefix="/viewp/.*"/>
        <data android:scheme="https" android:host="www.myapp.in" android:pathPrefix="/posts/.*"/>
        <data android:scheme="http" android:host="www.myapp.in"/>
        <data android:scheme="https" android:host="www.myapp.in"/>
        <category android:name="android.intent.category.BROWSABLE"/>
        <category android:name="android.intent.category.DEFAULT"/>
      </intent-filter>
    </activity>

**Comment by DSp4wN** [comment]
Please help me over this

**Comment by brentvatne** [comment]
@DSp4wN - you may be able to get help on discord: https://chat.expo.dev/

github issues are for demonstrable bugs, it seems like in your case there is likely just some missing piece of configuration. i'm not very familiar with this particular configuration myself so i can't help there
